lib/elixir/lib/module.ex

```diff
@@ -1726,7 +1726,16 @@ defmodule Module do
   def get_attribute(module, key, line) when is_atom(key) do
     assert_not_compiled!(:get_attribute, module)
     {set, bag} = data_tables_for(module)
+    get_attribute(module, key, line, set, bag)
+  end
+
+  defp get_attribute(_module, :spec, _line, _set, bag) do
+    specs = bag_lookup_element(bag, :spec, 2)
```

Comment:
Hrm, I am wondering. Maybe all we need to do is to add `spec` and friends as an accumulate attribute here? https://github.com/elixir-lang/elixir/blob/343b65c112fc8809d9b7445c92d7a403cdcdab09/lib/elixir/src/elixir_module.erl#L265

Reply:
Nah, it won't work because all accumulate attributes are stored as `{accumulate, Key}`. I am wondering if it is worth to store spec/type/callback and friends as `{accumulate, Key}` but the main issue is that type has a different return type and it won't work transparently for @type, @opaque and @typep.

Reply:
Oh! I was only focusing on `@spec`, but I see it was working previously with the rest of attributes (accumulating). 

> but the main issue is that type has a different return type and it won't work transparently for @type, @opaque and @typep.

I think we can make it work if we change `translate_typespecs_for_module/2` to get `:type`, `:typep` and `:opaque` from `bag` and adjust the rest of the functions to use `{kind, {expr, pos}}` instead of `{:type, {kind, expr, pos}}`. What do you think?